llvm/include/llvm/ADT/APInt.h

```diff
@@ -2187,6 +2187,11 @@ inline const APInt &umax(const APInt &A, const APInt &B) {
   return A.ugt(B) ? A : B;
 }
 
+/// Determine the absolute difference of two APInts considered to be unsigned.
+inline const APInt absdiff(const APInt &A, const APInt &B) {
+  return umax(A, B) - umin(A, B);
```

Comment:
I think it would be better to implement this in terms of `A >= B ? A - B : B - A`.

Reply:
Sure no problem, is this purely for ease of understanding or anything else?

Reply:
I was thinking it would be marginally more efficient.